### ISSUE TITLE
the deployment problem of O.W, O.F

### ISSUE THREAD

**Issue by Mr-Da-Yang** [wiwjp/ABD-Net#40]
First of all thank you very much for your outstanding contribution！
In the paper, you have deployed O.M and O.F many times. I have been searching for a long time and have not found the deployment py file for deploying O.M and O.F. Can you give me some guidance?
best wishs

**Comment by young169** [comment]
> First of all thank you very much for your outstanding contribution！
> In the paper, you have deployed O.M and O.F many times. I have been searching for a long time and have not found the deployment py file for deploying O.M and O.F. Can you give me some guidance?
> best wishs

I think it's in the torchreid/models/resnet.py for example. there is a 'ShallowCAM' imported from 'torchreid.components.shallow_cam'.
And have you run the training as well as testing both successfully?

**Comment by Mr-Da-Yang** [comment]
1. ShallowCAM is the attention module，O.M is a kind of regularization, which should appear in the process of network backpropagation, but it has not been found yet
2.  Currently not running this project

**Comment by shuaizzZ** [comment]
@Mr-Da-Yang 
        if not fixbase:
            reg = regularizer(model)  # ow
            loss += reg
        if not fixbase and args.use_of and epoch >= args.of_start_epoch:

            penalty = of_penalty(outputs)  # of
            loss += penalty

**Comment by Mr-Da-Yang** [comment]
> @Mr-Da-Yang
> if not fixbase:
> reg = regularizer(model) # ow
> loss += reg
> if not fixbase and args.use_of and epoch >= args.of_start_epoch:
> 
> ```
>         penalty = of_penalty(outputs)  # of
>         loss += penalty
> ```

Thank you very much!
I get it.

**Comment by shuaizzZ** [comment]
@Mr-Da-Yang 
方便交流一下吗，我也刚刚开始学习这篇文章，shuaizzz@yeah.net
